wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/DevFileConverter.java

```diff
@@ -0,0 +1,199 @@
+    config.setCommands(commands);
+    return config;
+  }
+
+  private List<CommandImpl> devCommandToCommandImpls(Devfile devFile, Command devCommand) {
+    List<CommandImpl> commands = new ArrayList<>();
+    for (Action devAction : devCommand.getActions()) {
+      CommandImpl command = new CommandImpl();
+      command.setName(devCommand.getName() + ":" + devAction.getTool());
+      command.setType(devAction.getType());
+      command.setCommandLine(devAction.getCommand());
+      if (devAction.getWorkdir() != null) {
+        command.getAttributes().put("workingDir", devAction.getWorkdir());
+      }
+      Optional<Tool> toolOfCommand =
+          devFile
+              .getTools()
+              .stream()
+              .filter(tool -> tool.getName().equals(devAction.getTool()))
+              .findFirst();
+      if (toolOfCommand.isPresent() && !isNullOrEmpty(toolOfCommand.get().getId())) {
+        command.getAttributes().put("pluginId", toolOfCommand.get().getId());
+      }
+      if (devCommand.getAttributes() != null) {
+        command.getAttributes().putAll(devCommand.getAttributes());
+      }
+      commands.add(command);
+    }
+    return commands;
+  }
+
+  private Command commandImplToDevCommand(CommandImpl command) {
+    Command devCommand = new Command().withName(command.getName());
+    Action action = new Action().withCommand(command.getCommandLine()).withType(command.getType());
+    if (!isNullOrEmpty(command.getAttributes().get("workingDir"))) {
+      action.setWorkdir(command.getAttributes().get("workingDir"));
+    }
+    // Remove internal attributes
+    command.getAttributes().remove("workingDir");
+    command.getAttributes().remove("pluginId");
+    // Put others
+    if (devCommand.getAttributes() == null) {
+      devCommand.withAttributes(command.getAttributes());
+    } else {
+      devCommand.getAttributes().putAll(command.getAttributes());
+    }
+    return devCommand;
+  }
+
+  private Project projectConfigToDevProject(ProjectConfigImpl projectConfig) {
+    Source source =
+        new Source()
+            .withType(projectConfig.getSource().getType())
+            .withLocation(projectConfig.getSource().getLocation());
+    return new Project().withName(projectConfig.getName()).withSource(source);
+  }
+
+  private ProjectConfigImpl devProjectToProjectConfig(Project devProject) {
+    ProjectConfigImpl projectConfig = new ProjectConfigImpl();
+    projectConfig.setName(devProject.getName());
+    projectConfig.setPath("/" + projectConfig.getName());
+    SourceStorageImpl sourceStorage = new SourceStorageImpl();
+    sourceStorage.setType(devProject.getSource().getType());
+    sourceStorage.setLocation(devProject.getSource().getLocation());
+    projectConfig.setSource(sourceStorage);
+    return projectConfig;
+  }
+
+  private String findToolName(WorkspaceConfigImpl wsConfig, String toolId) {
+    String aliasesString = firstNonNull(wsConfig.getAttributes().get("toolsAliases"), "");
```

Comment:
Please move `"toolsAliases"` somewhere to constant and describe where it is supposed to be used and describe format there. I think Constants in dev file module is good place